src/reader/decode/rle_v2.rs

```diff
@@ -2,86 +2,67 @@ pub mod delta;
-use std::io::{ErrorKind, Read};
+use std::{collections::VecDeque, io::Read};
 
-use crate::error::{self, Result};
+use crate::error::Result;
+
+use super::util::try_read_u8;
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-enum EncodingTypeV2 {
+enum EncodingType {
     ShortRepeat,
     Direct,
     PatchedBase,
     Delta,
 }
 
-#[inline]
-fn run_encoding(header: u8) -> EncodingTypeV2 {
-    match (header & 128 == 128, header & 64 == 64) {
-        // 11... = 3
-        (true, true) => EncodingTypeV2::Delta,
-        // 10... = 2
-        (true, false) => EncodingTypeV2::PatchedBase,
-        // 01... = 1
-        (false, true) => EncodingTypeV2::Direct,
-        // 00... = 0
-        (false, false) => EncodingTypeV2::ShortRepeat,
+impl EncodingType {
+    /// Checking highest two bits for encoding type.
+    #[inline]
+    fn from_header(header: u8) -> Self {
+        match (header & 128 == 128, header & 64 == 64) {
+            // 11... = 3
+            (true, true) => Self::Delta,
+            // 10... = 2
+            (true, false) => Self::PatchedBase,
+            // 01... = 1
+            (false, true) => Self::Direct,
+            // 00... = 0
+            (false, false) => Self::ShortRepeat,
+        }
     }
 }
 
 pub struct RleReaderV2<R> {
     reader: R,
     signed: bool,
-    literals: Vec<i64>,
-    num_literals: usize,
-    used: usize,
-    skip_corrupt: bool,
+    literals: VecDeque<i64>,
 }
 
-const MAX_SCOPE: usize = 512;
+const MAX_RUN_LENGTH: usize = 512;
 
 impl<R: Read> RleReaderV2<R> {
-    pub fn new(reader: R, signed: bool, skip_corrupt: bool) -> Self {
+    pub fn new(reader: R, signed: bool) -> Self {
         Self {
             reader,
-            num_literals: 0,
             signed,
-            literals: vec![0; MAX_SCOPE],
-            used: 0,
-            skip_corrupt,
+            literals: VecDeque::with_capacity(MAX_RUN_LENGTH),
         }
     }
 
     // returns false if no more bytes
-    pub fn read_values(&mut self, ignore_eof: bool) -> Result<bool> {
-        let mut byte = [0u8];
-
-        if let Err(err) = self.reader.read_exact(&mut byte) {
-            return match err.kind() {
-                ErrorKind::UnexpectedEof => {
-                    if !ignore_eof {
-                        Ok(false)
-                    } else {
-                        error::UnexpectedSnafu {
-                            msg: "Read past end of RLE integer reader",
-                        }
-                        .fail()
-                    }
-                }
-                _ => error::UnexpectedSnafu {
-                    msg: err.to_string(),
-                }
-                .fail(),
-            };
-        }
-
-        let header = byte[0];
-        let encoding = run_encoding(header);
+    pub fn read_values(&mut self) -> Result<bool> {
+        let header = match try_read_u8(&mut self.reader)? {
+            Some(byte) => byte,
+            None => return Ok(false),
+        };
```

Comment:
Removed the ignore_eof logic, as have similar thoughts to the removal of skip corrupt